src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/ElementsMarshalling.cs

```diff
@@ -315,6 +315,33 @@ protected StatementSyntax GenerateByValueOutUnmarshalStatement(TypePositionInfo
                     StubCodeContext.Stage.Unmarshal));
         }
 
+        protected StatementSyntax GenerateElementCleanupStatement(TypePositionInfo info, StubCodeContext context)
+        {
+            string nativeSpanIdentifier = MarshallerHelpers.GetNativeSpanIdentifier(info, context);
+            StatementSyntax contentsCleanupStatements = GenerateContentsMarshallingStatement(info, context,
+                    MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression,
+                        IdentifierName(MarshallerHelpers.GetNativeSpanIdentifier(info, context)),
+                        IdentifierName("Length")),
+                        StubCodeContext.Stage.Cleanup);
+
+            if (!contentsCleanupStatements.IsKind(SyntaxKind.EmptyStatement))
```

Comment:
Can we flip this predicate instead? The inner block is the meat of the logic we want to execute, not the empty statement.